iceoryx_posh/include/iceoryx_posh/internal/roudi/memory/mempool_collection_memory_block.hpp

```diff
@@ -45,21 +45,20 @@ class MemPoolCollectionMemoryBlock final : public MemoryBlock
     MemPoolCollectionMemoryBlock& operator=(const MemPoolCollectionMemoryBlock&) = delete;
     MemPoolCollectionMemoryBlock& operator=(MemPoolCollectionMemoryBlock&&) = delete;
 
-    /// @brief Implementation of MemoryBlock::size
-    /// @return the size of type T
+    /// @copydoc MemoryBlock::size
+    /// @note The size for all the MemPools
     uint64_t size() const noexcept override;
 
-    /// @brief Implementation of MemoryBlock::alignment
-    /// @return the alignment of type T
+    /// @copydoc MemoryBlock::alignment
+    /// @note The memory alignment for the MemPools
     uint64_t alignment() const noexcept override;
 
-    /// @brief Implementation of MemoryBlock::memoryAvailable
-    /// This will create the MemPools
-    /// @param [in] memory pointer to a valid memory location to place the mempools
+    /// @copydoc MemoryBlock::memoryAvailable
+    /// @note This will create the MemPools at the location `memory` points to
     void memoryAvailable(void* memory) noexcept override;
```

Comment:
Is `nullptr` as argument allowed? If not please use `cxx::not_null<void>` as argument here, in the base class and all the children.

Reply:
You are right and if possible this would be a reference. The method should also not be public in the first place. I would like to fix this in a follow up PR .

Reply:
done in #843